### PR TITLE
feat: Add Android documentation for logging plugin

### PR DESCRIPTION
**Pull request by sdhuka**
#### Description of changes:
* Add android changes for central logging plugin.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [X] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [X] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [X] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [X] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [X] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/logging/android/escapehatch/escapehatch.mdx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+```java
+AWSCloudWatchLoggingPlugin plugin = (AWSCloudWatchLoggingPlugin)Amplify.Logging.getPlugin("awsCloudWatchLoggingPlugin");
+CloudWatchLogsClient client = plugin.getEscapeHatch();
+```
+
+</Block>
+<Block name="Kotlin">
+
+```kotlin
+val plugin = Amplify.Logging.getPlugin("") as AWSCloudWatchLoggingPlugin
```
Comment (tjleing) [inline src/fragments/lib/logging/android/escapehatch/escapehatch.mdx:25]: does this work or do you need to pass the plugin name?

File: src/fragments/lib/logging/android/flush_logs/flush_logs.mdx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }, error -> {
+        // failed to flush logs
+    }
+);
+```
+</Block>
+<Block name="Kotlin">
+
+Add import statement to the `AWSCloudWatchLoggingPlugin`
+```java
```
Comment (tjleing) [inline src/fragments/lib/logging/android/flush_logs/flush_logs.mdx:24]:
```suggestion
```kotlin
```

File: src/fragments/lib/logging/android/flush_logs/flush_logs.mdx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import com.amplifyframework.logging.cloudwatch.AWSCloudWatchLoggingPlugin
+```
+
+Execute the flush log function from the plugin.
+
+```java
```
Comment (tjleing) [inline src/fragments/lib/logging/android/flush_logs/flush_logs.mdx:30]:
```suggestion
```kotlin
```

File: src/fragments/lib/logging/android/common/10_category_list.mdx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+The following are existing Amplify category names that are used by default by Amplify when automatically logging errors from the library.
```
[the remark this answers is not in the document]
Comment (sdhuka) [reply src/fragments/lib/logging/android/common/10_category_list.mdx:1]: this clarification should be part of the predictions category documentation.

**Review by tjleing — APPROVED** [review]

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

**Pull request merged**